packages/utils/lib/utils.ts

```diff
@@ -323,4 +323,22 @@ export async function extractZip(zip: AdmZip, dest: string, overwrite = false, s
     }
 }
 
+export async function pipeRequest(req: superagent.Request, w: fs.WriteStream, timeout?: number, name?: string) {
+    await req.buffer(false).parse((resp, cb) => {
```

Comment:
Use `await` here will wait for request to finish and totally ignore the timeout promise part at L331. I would suggest to use `.timeout()` functionality provided by `super agent` instead. For example,

```javascript
req.buffer(false).timeout(timeout)....
```